doc/src/suite-config.rst

```diff
@@ -0,0 +1,3772 @@
+
+A typical example:
+
+.. code-block:: bash
+
+   /path/to/my/suite   # suite configuration directory
+       suite.rc           # THE SUITE CONFIGURATION FILE
+       bin/               # scripts and executables used by tasks
+           foo.sh
+           bar.sh
+           ...
+       # (OPTIONAL) any other suite-related files, for example:
+       inc/               # suite.rc include-files
+           nwp-tasks.rc
+           globals.rc
+           ...
+       doc/               # documentation
+       control/           # control files
+       ancil/             # ancillary files
+       ...
+
+
+.. _SuiteRCFile:
+
+Suite.rc File Overview
+----------------------
+
+Suite.rc files are an extended-INI format with section nesting.
+
+Embedded template processor expressions may also be used in the file, to
+programatically generate the final suite configuration seen by
+cylc. Currently the `Jinja2 <http://jinja.pocoo.org/docs>`_ and
+`EmPy <http://www.alcyone.com/software/empy>`_ template processors are
+supported; see :ref:`Jinja` and :ref:`EmPylabel` for examples. In the future
+cylc may provide a plug-in interface to allow use of other template
+engines too.
+
+
+.. _Syntax:
+
+Syntax
+^^^^^^
+
+The following defines legal suite.rc syntax:
+
+- **Items** are of the form ``item = value``.
+- **[Section]** headings are enclosed in square brackets.
+- **Sub-section [[nesting]]** is defined by repeated square brackets.
+
+  - Sections are **closed** by the next section heading.
+
+- **Comments** (line and trailing) follow a hash character: ``#``
+- **List values** are comma-separated.
+- **Single-line string values** can be single-, double-, or un-quoted.
+- **Multi-line string values** are triple-quoted (using
+  single or double quote characters).
+- **Boolean values** are capitalized: True, False.
+- **Leading and trailing whitespace** is ignored.
+- **Indentation** is optional but should be used for clarity.
+- **Continuation lines** follow a trailing backslash: ``\``
+- **Duplicate sections** add their items to those previously
+  defined under the same section.
+- **Duplicate items** override, *except for dependency
+  ``graph`` strings, which are additive*.
+- **Include-files** ``%include inc/foo.rc`` can be
+  used as a verbatim inlining mechanism.
+
+Suites that embed templating code (see :ref:`Jinja` and :ref:`EmPylabel`) must
+process to raw suite.rc syntax.
+
+
+Include-Files
+^^^^^^^^^^^^^
+
+Cylc has native support for suite.rc include-files, which may help to
+organize large suites. Inclusion boundaries are completely arbitrary -
+you can think of include-files as chunks of the suite.rc file simply
+cut-and-pasted into another file. Include-files may be included
+multiple times in the same file, and even nested. Include-file paths
+can be specified portably relative to the suite configuration directory,
+e.g.:
+
+.. code-block:: cylc
+
+   # include the file $CYLC_SUITE_DEF_PATH/inc/foo.rc:
+   %include inc/foo.rc
```

Comment:
This code-block gets rendered in red as an error, this is my fault (sorry) I didn't add the include file syntax to the `cylc_lang` lexer. These errors can be ignored in this PR.